### PR TITLE
Show diagnostic ids in analyzer report

### LINKED ISSUE #51695: Print DiagnosticIds in the Report Analyzer Format
Hello.

This new feature suggestion is related to the [Report Analyzer Format](https://github.com/dotnet/roslyn/blob/main/docs/analyzers/Report%20Analyzer%20Format.md), which can be obtained by running the build with the `/v:d /p:reportanalyzer=true` parameters.

In SonarSource, we often ask to our users to share verbose logs with `/p:ReportAnalyzer` to generate the report with time spent by each analyzer.

One problem we have is that in the output, the analyzer class name is written, and it's not easy to map this to the supported diagnosticId(s). When providing support, it is useful to know the diagnosticId(s), too.

We would like to ask if there is a recommended way to include the rule ID in the `/p:ReportAnalyzer` logs, or if it would be feasible to be done on the Roslyn side. We are aware the multiple DiagnosticIds can be assigned to an Analyzer, and this feature would need to handle this scenario as well.

ACTUAL:
```
  Time (s)    %   Analyzer
     1.352   87   SonarAnalyzer.CSharp, Version=8.19.0.0, Culture=neutral, PublicKeyToken=c5b62af9de6d7244
     0.110    7      SonarAnalyzer.Rules.CSharp.UnusedReturnValue
     0.039    2      SonarAnalyzer.Rules.CSharp.UnusedPrivateMember
```
EXPECTED:
```
  Time (s)    %   Analyzer
     1.352   87   SonarAnalyzer.CSharp, Version=8.19.0.0, Culture=neutral, PublicKeyToken=c5b62af9de6d7244
     0.110    7      SonarAnalyzer.Rules.CSharp.UnusedReturnValue (S3241)
     0.039    2      SonarAnalyzer.Rules.CSharp.UnusedPrivateMember (S1144, S4487)
```
In the above example, the analyzer `UnusedReturnValue` has only one diagnosticId, whereas the `UnusedPrivateMember` analyzer has two diagnosticIds.

Thanks.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #51695

Waiting for CI to tell me where the tests are to updated them :smile:

### CODE REVIEW

File: src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs [hunk 1]
```diff
@@ -1449,7 +1449,7 @@ private static void ReportAnalyzerExecutionTime(TextWriter consoleOutput, Analyz
 
                     analyzerTimeColumn = getFormattedTime(executionTime);
                     analyzerPercentageColumn = getFormattedPercentage(percentage);
-                    analyzerNameColumn = getFormattedAnalyzerName("   " + kvp.Key.ToString());
+                    analyzerNameColumn = getFormattedAnalyzerName($"   {kvp.Key} ({string.Join(", ", kvp.Key.SupportedDiagnostics.Select(d => d.Id))})");
```
Comment (sharwell) [inline src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:1452]: 💡 Can you make the list of IDs distinct and ordered?

Comment (Youssef1313) [reply src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:1452]: Any concerns doing much LINQ in this code path?

Also the CI is passing. So it looks like this wasn't tested. I'm not aware of how to test this 😕

Comment (sharwell) [reply src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:1452]: > Any concerns doing much LINQ in this code path

Should be fine

> I'm not aware of how to test this

@mavasani ?

**Review by jmarolf — APPROVED** [review]

File: src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs (hunk 1 above)
Comment (cston) [reply src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:1452]: > I'm not aware of how to test this

It looks like we have a few tests that expect analyzer execution time such as [`ReportAnalyzerOutput`](https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs#L9143). Perhaps a similar test there?

Comment (Youssef1313) [reply src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:1452]: @cston Thanks for the pointer. I think I could just update the same test.

**Review by cston — APPROVED** [review]

**Pull request merged**